app/controllers/users/viewing_parties_controller.rb

```diff
@@ -1,5 +1,67 @@
 class Users::ViewingPartiesController < ApplicationController
+  before_action :load_host
+  before_action :load_movie
+  before_action :validate_duration, only: [:create]
+
   def new
+    @all_users = User.where.not(id: @host)
+  end
+
+  def create
+    viewing_party = ViewingParty.new(viewing_party_params)
+
+    if viewing_party.save
+      viewing_party_user_ids.each do |user_id|
+        viewing_party.users << User.find(user_id)
+      end
+
+      redirect_to user_path(@host.id)
+    else
+      redirect_to new_user_movie_viewing_party_path(@host.id, @movie.id)
+
+      flash[:alert] = viewing_party.errors.full_messages.to_sentence
+    end
+  end
+
+  private
+
+  def load_host
+    @host ||= User.find(params[:user_id])
+  end
+
+  def load_movie
+    @movie ||= MovieFacade.new(params[:movie_id]).movie
+  end
+
+  def validate_duration
+    if viewing_party_params[:duration].to_i < @movie.runtime
+      flash[:alert] = "Viewing Party duration must be greater than or equal to movie runtime which is #{@movie.runtime}"
+      redirect_to new_user_movie_viewing_party_path(@host.id, @movie.id)
+    end
+  end
+
+  def date
+    "#{params["date(1i)"]}/#{params["date(2i)"]}/#{params["date(3i)"]}"
+  end
+
+  def time 
+    "#{params["start_time(4i)"]}:#{params["start_time(5i)"]}:00"
+  end
+
```

Comment:
I don't understand the syntax here, I assume it is has something to do with the way the form passes date/time objects via params